tests/h/models/user_test.py

```diff
@@ -7,175 +7,176 @@
 from h import models
 
 
-def test_cannot_create_dot_variant_of_user(db_session):
-    fred = models.User(authority='example.com',
-                       username='fredbloggs',
-                       email='fred@example.com')
-    fred2 = models.User(authority='example.com',
-                        username='fred.bloggs',
-                        email='fred@example.org')
-
-    db_session.add(fred)
-    db_session.add(fred2)
-    with pytest.raises(exc.IntegrityError):
+class TestUserModelDataConstraints(object):
+    """Unit tests for :py:module:`User` data integrity constraints"""
```

Comment:
I think this is a `:py:class` not a `module`? And it should probably be `h.models.User`. Or maybe just `models.User` would work, not sure.

Reply:
Great question. For the past few months, finding actual documentation for how to write this stuff has been elusive. Sphinx...I guess?

Reply:
So, `:py:module:`h.models.User`` ? Or do I have it way wrong?

Reply:
:py:class:`h.models.User`

They're the [Sphinx Python domain](http://www.sphinx-doc.org/en/master/usage/restructuredtext/domains.html#the-python-domain).

If we used [Sphinx autodoc](http://www.sphinx-doc.org/en/master/ext/autodoc.html) to pull our docstrings out of our code and into our Sphinx / readthedocs docs, then we could have a `make docs` that errors when one of these is wrong (including when a future refactoring breaks a reference in a docstring). But we don't have that in place